instrumentation-core/spring/spring-core/src/main/java/io/opentelemetry/spring/instrumentation/httpclients/RestTemplateInterceptorTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.spring.instrumentation.httpclients;
+
+import io.grpc.Context;
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.Span;
+import io.opentelemetry.trace.Tracer;
+import java.io.IOException;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.ClientHttpRequestExecution;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpResponse;
+
+/** Wraps RestTemplate requests in a span. Adds the current span context to request headers. */
+public final class RestTemplateInterceptorTest implements ClientHttpRequestInterceptor {
```

Comment:
This class should not be named `Test`. Also, this is an interceptor that uses opentelemetry, so better name is `OpenTelemetryRequestInterceptor`

Reply:
Yes, this was a mistake. I though I renamed a test and not the main java file. I'll fix this